__tests__/__unit__/dataset/actions.unit.test.ts

```diff
@@ -1584,3 +1588,383 @@ describe("Dataset Actions Unit Tests - Function hRecallDataSet", () => {
+        const profileInstance = createInstanceOfProfile(imperativeProfile);
+        const zosmfSession = createBasicZosmfSession(imperativeProfile);
+        const treeView = createTreeView();
+        const datasetSessionNode = createDatasetSessionNode(session, imperativeProfile);
+        const testDatasetTree = createDatasetTree(datasetSessionNode, treeView);
+        const mvsApi = createMvsApi(imperativeProfile);
+        bindMvsApi(mvsApi);
+
+        return {
+            session,
+            sessionWithoutCredentials,
+            zosmfSession,
+            treeView,
+            imperativeProfile,
+            datasetSessionNode,
+            mvsApi,
+            profileInstance,
+            testDatasetTree
+        };
+    }
+
+    it("Checking of proper configuration being picked up for different DS types", async () => {
+        createGlobalMocks();
+        const blockMocks = createBlockMocks();
+
+        blockMocks.testDatasetTree.createFilterString.mockResolvedValue("test");
+        blockMocks.testDatasetTree.getHistory.mockReturnValue([]);
+        mocked(Profiles.getInstance).mockReturnValue(blockMocks.profileInstance);
+        mocked(vscode.window.showInputBox).mockResolvedValue("test");
+        const createDataSetSpy = jest.spyOn(blockMocks.mvsApi, "createDataSet");
+        createDataSetSpy.mockReset();
+        const getChildrenSpy = jest.spyOn(blockMocks.datasetSessionNode, "getChildren");
+        getChildrenSpy.mockResolvedValue([]);
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Binary" as any);
+        await dsActions.createFile(blockMocks.datasetSessionNode, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-Binary");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set C" as any);
+        await dsActions.createFile(blockMocks.datasetSessionNode, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-C");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Classic" as any);
+        await dsActions.createFile(blockMocks.datasetSessionNode, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-Classic");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Partitioned" as any);
+        await dsActions.createFile(blockMocks.datasetSessionNode, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-PDS");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Sequential" as any);
+        await dsActions.createFile(blockMocks.datasetSessionNode, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-PS");
```

Comment:
These area really 5 different tests. It would make sense to separate them. But since @stepanzharychevbroadcom has moved to another group and this is an improvement over what we had, lets leave it like this and restructure it in the future.

Reply:
Stepan's response to the issue was as follows:

> Answering on all at once, specific of this test is verification of proper run with different types without details, splitting it will result in hundreds of lines of the duplicates. I think that's an exception which can be made, if no - fine, I will spend time splitting it to 5 test cases with duplicates.

Reply:
I think that is a bit of exaggerated. The setup logic can be share in a describe block or extracted to function and reused.